test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,11 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        //Check only when file is mutated
+        if (inputCommand.equals("add") || inputCommand.equals("clear") ||
+                inputCommand.equals("delete")) {
+            assertEquals(addressBook, saveFile.load());
+        }
```

Comment:
Instead of listing out the mutating commands explicitly, you should use isMutating method to check whether it is a mutating command. Also, to be complete, you should have an `else` branch that confirms the file was not updated for non mutating commands (e.g. by checking the last updated time of the file).